### PR TITLE
Add Http-service to check bookie sanity state

### PR DESCRIPTION
**Pull request by rdhabalia**
### Motivation

Right now, if Bookie startup or process fails due to various reasons then bookie getState() rest API still gives a successful response even though bookie writes are not going through. Therefore. Bookie should have a REST API to check bookie sanity test and that can be used to check bookie liveliness in k8/helm-chart or to plot bookie health graph.

### Changes

Add Bookie REST API to check bookie health to validate bookie liveliness

### CODE REVIEW

**Comment by rdhabalia** [comment]
good point. made a change to run max (1) concurrent requests.

**Comment by rdhabalia** [comment]
@eolivelli can you PTAL.

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/server/http/service/BookieSanityService.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		}
+
+		BookieSanity bs = new BookieSanity();
+		if (config.isForceReadOnlyBookie()) {
+			bs.readOnly = true;
+		} else {
+		    try {
+                // allow max concurrent request as sanity-test check relatively
+                // longer time to complete
+		        lock.acquire();
```
Comment (eolivelli) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/server/http/service/BookieSanityService.java:87]: Please wait at most for X seconds (5?) and if the semaphore cannot be acquired fail the request. We don't want many requests piling up.


Please also fix formatting

**Review by eolivelli — APPROVED** [review]
I have left one minor issue please check my comment.
Then the patch is good to go from my POV.
And we should release this with 4.16.

@hangc0276 FYI

**Comment by hangc0276** [comment]
> I have left one minor issue please check my comment. Then the patch is good to go from my POV. And we should release this with 4.16.
> 
> @hangc0276 FYI

@eolivelli Sure

**Review by hangc0276 — APPROVED** [review]
Overall LGTM.

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/server/http/service/BookieSanityService.java (hunk 1 above)
Comment (rdhabalia) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/server/http/service/BookieSanityService.java]: done

**Pull request merged**